### PR TITLE
feat: adding max_count parameter to limit posts on download

### LINKED ISSUE #2269: get_posts() with limit
Hi. Make the **get_posts()** method accept the limit on the received data as an argument. There are pages with more than 100 thousand posts and it took me about an hour to do this.
I would like to get only the last n posts instead of all the posts by sending an argument. 
`posts = user.get_posts(50); #latest 50 posts only :)`

### PR DESCRIPTION
**Pull request by Mesheo**
### Motivation for this change
👋 Hi! I made this PR to make it possible to control the quantity of posts downloaded by profile.
The core method `self.posts_download_loop()` already has this functionality, but the external calling function `instaloader.download_profiles()` were not adjusted to receive or pass this argument further.

  - Fixes #2269. (I believe this change will solve this issue) 

### Changes proposed in this pull request
* Added parameter `max_count` to `download_profiles()` method

### The completeness of this change
  - Is it just a proof of concept? **No**
  - Is the documentation updated (if appropriate)? **Yes**
  - Do you consider it ready to be merged or is it a draft? **Ready** 
  - Can we help you at some point? **Feel free to give me some guidelines to better solve this problem, I am fully available to do so.**

### CODE REVIEW

**Comment by Mesheo** [comment]
I believe this new PR addresses the past comment by @aandergr 
* https://github.com/instaloader/instaloader/pull/2271#issuecomment-2138870438 
Let me know if need any adjustments

File: instaloader/instaloader.py [hunk 1]
```diff
@@ -1434,6 +1435,9 @@ def download_profiles(self, profiles: Set[Profile],
 
         .. versionchanged:: 4.8
            Add `latest_stamps` parameter.
+
+        .. versionchanged:: 4.12
+           Add `max_count` parameter.
```
Comment (aandergr) [inline instaloader/instaloader.py:1440]: Would you please also describe the parameter in the :param: listing here in the method docstring?

Comment (Mesheo) [reply instaloader/instaloader.py:1441]: @aandergr

File: instaloader/instaloader.py [hunk 2]
```diff
@@ -1426,6 +1427,7 @@ def download_profiles(self, profiles: Set[Profile],
            Whether :exc:`LoginRequiredException` and :exc:`PrivateProfileNotFollowedException` should be raised or
            catched and printed with :meth:`InstaloaderContext.error_catcher`.
         :param latest_stamps: :option:`--latest-stamps`.
+        :param max_count: Maximum count of posts to download
```
Comment (aandergr) [inline instaloader/instaloader.py:1430]:
```suggestion
        :param max_count: Maximum count of posts to download.
```

**Comment by aandergr** [comment]
Thanks!

**Pull request merged**